auth/service/api/v1/oauth.go

```diff
@@ -104,20 +107,48 @@ func (r *Router) OAuthProviderRedirectGet(res rest.ResponseWriter, req *rest.Req
 		return
 	}
 
+	redirectURL := path.Join(req.BaseUrl().String(), prvdr.Type(), prvdr.Name())
+	redirectURLAuthorized := path.Join(redirectURL, "authorized")
+	redirectURLDeclined := path.Join(redirectURL, "declined")
+
 	restrictedToken, err := r.oauthProviderRestrictedToken(req.Request, prvdr)
 	if err != nil {
 		r.htmlOnError(res, req, err)
 		return
 	}
 
+	// Include custodial account signup credentials in redirect URL query, if applicable
+	confirmations, err := r.ConfirmationClient().GetAccountSignupConfirmationWithResponse(ctx, confirmationClient.UserId(restrictedToken.UserID))
+	if err != nil {
+		r.htmlOnError(res, req, err)
+		return
+	}
+
+	signupParams := url.Values{}
+	if confirmations.JSON200 != nil && len(confirmationClient.ConfirmationList(*confirmations.JSON200)) > 0 {
+		confirmation := confirmationClient.ConfirmationList(*confirmations.JSON200)[0]
```

Comment:
Do you need to check the confirmation status here? Can `hydrophone` return sign up confirmations that are already completed?

Reply:
I think this is OK, since hydrophone filters the query for `pending` status:
https://github.com/tidepool-org/hydrophone/blob/e398604f78b07f2c154b4899b379f762af7d81da/api/signup.go#L426